lib/puppet/scheduler/splay_job.rb

```diff
@@ -25,6 +25,14 @@ def ready?(time)
       end
     end
 
+    # Recalculates the splay limit.
+    #
+    # @param splay_limit [Integer] the maximum time to delay before an agent's first run.
+    # @return [Integer] a random integer less than or equal to the splay limit.
+    def splay_limit=(splay_limit)
+      @splay = calculate_splay(splay_limit)
+    end
+
```

Comment:
Could you add a test that it's possible to set the new limit and the resulting `splay` is recalculated? A common pattern we use in puppet is something like:

```ruby
now = Time.now
allow(Time).to receive(:now).and_return(now)
```

That way if you can writer deterministic expectations like
```ruby
expect(job.splay).to eq(now + 100)
```

If you need to be more lenient, you can do things like:

```ruby
expect(..).to be_within(1.second).of(now + 100)
expect(..).to be_between(min, max)
```

Reply:
Maybe when the `reparse_run` job runs in the daemon, verify the `agent_run` "next time to run" is different than before, something like

```ruby
      Puppet[:filetimeout] = 1 # this is needed so the `reparse_run` job isn't disabled
      Puppet[:splay] = true
      allow(agent).to receive(:run)

      daemon.start
      now = Time.now
      next = scheduler.jobs[0].interval_to_next_from(now)
      # force the `reparse_run` job to run
      scheduler.jobs[1].run(now)
      # might want to stub `rand` so the recalculated splay time is deterministic and can never be the same as `next`?
      expect(scheduler.jobs[0].interval_to_next_from(now)).to_not eq(next)
```